packages/cli/src/lib/snapshot/snapshotReporter.ts

```diff
@@ -57,6 +89,110 @@ export class SnapshotReporter {
     );
   }
 
+  private reportHandlers: SnapshotReporterHandlers = {
+    [SnapshotReportStatus.SUCCESS]: () => {},
+    [SnapshotReportStatus.NO_CHANGES]: () => {},
+    [SnapshotReportStatus.MISSING_VAULT_ENTRIES]: () => {},
+    [SnapshotReportStatus.ERROR]: () => {},
+  };
+
+  public setReportHandler<T extends SnapshotReportStatus>(
+    status: T,
+    handler: SnapshotReporterHandlers[T]
+  ): SnapshotReporter {
+    this.reportHandlers[status] = handler;
+    return this;
+  }
+
+  public async handleReport(): Promise<void> {
+    const reportStatuses = this.getReportStatuses();
+    for (const handler of reportStatuses.fixables) {
+      await this.reportHandlers[handler].apply(this);
+      this.reportHandlers[handler] = () => {};
+    }
+    if (reportStatuses.errors.length > 0) {
+      for (const handler of reportStatuses.errors) {
+        await this.reportHandlers[handler].apply(this);
+        this.reportHandlers[handler] = () => {};
+      }
+    } else if (reportStatuses.successes.length > 0) {
```

Comment:
- [ ] UT ensures success is **not** called when there are errors.
- [ ] UT ensures success is called when there are no errors.